pkg/compactor/retention/expiration_test.go

```diff
@@ -540,3 +545,40 @@ func TestExpirationChecker_IntervalMayHaveExpiredChunks(t *testing.T) {
 		})
 	}
 }
+
+func Test_PolicyStreamMapping_PolicyFor(t *testing.T) {
```

Comment:
This is great! I think we should make it a bit more complete tho:
- Two mapping policies with the same priority: Get first one matchiong. We should log an error with insight logs and some metrics but we can do that in a followup PR (I added a task to the epic for this)
- Regexes
- Multiple mappings per policy

Reply:
Pushed a commit addressing those scenarios.